lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -285,6 +285,9 @@ def add_test_targets
                 # requires frameworks. For tests we always use the test target name as the product name
                 # irrelevant to whether we use frameworks or not.
                 configuration.build_settings['PRODUCT_NAME'] = name
+                # target_installer sets 'MACH_O_TYPE' for static frameworks ensure this does not propagate
+                # to test target.
+                configuration.build_settings.delete('MACH_O_TYPE')
```

Comment:
I really dislike that super class sets a bunch of these and we have to delete them but I won't refactor it for 1.6.0.

Reply:
agree this isn't ideal